src/fights.ts

```diff
@@ -323,157 +319,137 @@ function bestWitchessPiece() {
           }
-          withMacro(firstChainMacro(), () =>
-            fightSource.run({
-              macro: firstChainMacro(),
-            })
+
+          useFamiliar($familiar`Pocket Professor`);
+          goalMaximize(
+            Requirement.merge([
+              new Requirement(maximizeParameters, {
+                forceEquip: $items`Pocket Professor memory chip`,
+              }),
+              ...fightSource.requirements,
+            ])
           );
-          embezzlerLog.initialEmbezzlersFought +=
-            1 + get("_pocketProfessorLectures") - startLectures;
-        }
-        set("_garbo_meatChain", true);
-        postCombatActions();
-      }
 
-      startWandererCounter();
-
-      // SECOND EMBEZZLER CHAIN
-      if (have($familiar`Pocket Professor`) && !get<boolean>("_garbo_weightChain", false)) {
-        const startLectures = get("_pocketProfessorLectures");
-        const fightSource = getNextEmbezzlerFight();
-        if (!fightSource) return;
-        useFamiliar($familiar`Pocket Professor`);
-        const requirements = Requirement.merge([
-          new Requirement(["Familiar Weight"], {
-            forceEquip: $items`Pocket Professor memory chip`,
-          }),
-          ...fightSource.requirements,
-        ]);
-        maximizeCached(requirements.maximizeParameters, requirements.maximizeOptions);
-        if (
-          get("_pocketProfessorLectures") <
-          2 + Math.ceil(Math.sqrt(familiarWeight(myFamiliar()) + weightAdjustment()))
-        ) {
-          if (["Macrometeorite", "Powerful Glove"].includes(fightSource.name)) {
-            saberCrateIfDesired();
-          }
-          withMacro(secondChainMacro(), () =>
+          if (get("_pocketProfessorLectures") < pocketProfessorCopies()) {
+            const startLectures = get("_pocketProfessorLectures");
             fightSource.run({
-              macro: secondChainMacro(),
-            })
-          );
-          embezzlerLog.initialEmbezzlersFought +=
-            1 + get("_pocketProfessorLectures") - startLectures;
+              macro: macro,
+            });
+            embezzlerLog.initialEmbezzlersFought +=
+              1 + get("_pocketProfessorLectures") - startLectures;
+          }
+          set(property, true);
+          postCombatActions();
+          startWandererCounter();
         }
-        set("_garbo_weightChain", true);
-        postCombatActions();
       }
 
-      startWandererCounter();
-
       // REMAINING EMBEZZLER FIGHTS
       let nextFight = getNextEmbezzlerFight();
       while (nextFight !== null) {
         const startTurns = totalTurnsPlayed();
         if (have($skill`Musk of the Moose`) && !have($effect`Musk of the Moose`)) {
           useSkill($skill`Musk of the Moose`);
         }
-        if (["Macrometeorite", "Powerful Glove"].includes(nextFight.name)) {
+        if (nextFight.gregariousReplace) {
           saberCrateIfDesired();
         }
         withMacro(embezzlerMacro(), () => {
```

Comment:
We don't need this wrapper anymore

Reply:
to clarify, the withMacro() wrapper was for EmbezzlerFights that don't involve adventureMacro, like the photocopy or chateau painting. But because of the setup here, it wasn't updating embezzlerMacro at the right time. It's better & more elevant to include withMacro in the run() functions of those EmbezzlerFights, which we do now.